security/advancedtls/examples/examples_test.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
```

Comment:
I can barely read bash. Please ask someone else to take a look at this.

Reply:
I don't, either :) @dfawley Could you please take a look at this please?
This is just a simple copy from https://github.com/grpc/grpc-go/blob/master/examples/examples_test.sh, with the following minor modifications:
1. put the right test name and output in `EXAMPLES`, `EXPECTED_SERVER_OUTPUT`, and `EXPECTED_CLIENT_OUTPUT`
2. replace `cd ./examples` with `cd ./security/advancedtls/examples`
3. add a `sleep 2s` at the time after we start the server and before we initiate the client request(I added this mainly because otherwise, the test could sometimes fail locally with a server unavailable error. I didn't try on Travis, but when I ran other tests in main grpc it would sometimes fail as well. I am guessing my machine is a bit old that sometimes the server code needs more time to boot up)

Reply:
> add a sleep 2s at the time after we start the server and before we initiate the client request(I added this mainly because otherwise, the test could sometimes fail locally with a server unavailable error. I didn't try on Travis, but when I ran other tests in main grpc it would sometimes fail as well. I am guessing my machine is a bit old that sometimes the server code needs more time to boot up)

That seems strange.  Do you see these errors when running the other `examples_test.sh`?  If the client sends wait-for-ready RPCs then you shouldn't need to worry about this at all.

Reply:
Ah, it turned out that the tests failed in other `examples_test.sh` was due to a port is already used on my machine, instead of the unavailable error. So I think other scripts are good.
Just to confirm, the way to send `wait-for-ready` RPCs is through `grpc.WithBlock()`, right? I've updated that, removed the 2s sleep time, and ran the script for 100 times. No errors anymore. 
Could you please check if the script make sense now?

Reply:
> Just to confirm, the way to send wait-for-ready RPCs is through grpc.WithBlock(), right? I've updated that, removed the 2s sleep time, and ran the script for 100 times. No errors anymore.

No, `WithBlock` makes `Dial` wait until there is a ready connection.  `WaitForReady` means the RPC will wait until the channel is ready (however long and after however many failures that takes) before sending the RPC.  It shouldn't be necessary, but if you are seeing weird failures due to connectivity, it can help.  `WithBlock` also shouldn't be necessary; I'd remove it unless you can explain why it fixes things.

Reply:
OK, sure. I've changed to `WaitForReady` and it also works. Please take a look, thank you so much!

Reply:
OH, I thought you wouldn't need WFR either, as both of these options are actually unusual.  But I see there is a race at startup here: if the client `Dial`s before the server is `Listen`ing, then it will get an error.  In this case, either `WithBlock` or WFR RPCs should be fine.

Reply:
Great! Thanks for the confirmation!